### ISSUE TITLE
Git and GitHub as the backend?

### ISSUE THREAD

**Issue by jacobdadams** [agrc/metadata-asset-tracking-tool#1]
Using git as the backend for MATT
- metadata stored in github as flat files
- CSV files for other tables
  - unique GUID-like primary key for all tables that identifies datasets
  - pre-commit hook that sorts on the foreign key field to ensure sane diffs
- a table that maps layer keys to metadata file names
- actions on PR merges that trigger the appropriate updates so we don't have to manually build the sgid index, update metadata in all the places, etc.
  - and/or have a regularly-scheduled update script running nightly/weekly/whatever
  - Maybe a webhook that triggers a master script in Cloud Run/Cloud Functions?

## Pros

- all the benefits of git for versioning, branching, and change tracking
- GitHub tools for PRs, issues, and Actions
- Same toolchain as website (vscode/git/github)

## Cons

- Same toolchain as website could require training or hamper usage by those not comfortable with that toolchain
- Git isn't great with tabular data
- proper DBs have much better features and performance
- Have to manage layer keys manually when linking tables and metadata files (though maybe this can be automated somehow?)

Or maybe a git repo is best for the metadata and other textual data and a proper DB handles the tabular stuff, and a script integrates the two?

**Comment by steveoh** [comment]
Let's use the github discussion feature for these instead of issues.

**Comment by jacobdadams** [comment]
Since you felt strongly enough to close the issue for me, I would have appreciated it if you had also moved it to a discussion item. Otherwise it comes across like you don't care for my ideas simply because I made a mistake of where to put them :)

**Comment by steveoh** [comment]
I now see the button. I did look for it albeit in the wrong place. 

I didn't read your idea enough to have an opinion so I apologize that it came off that way. I care about everyones ideas!

I'm hoping we can have a larger discussion about this project as a group. I'll assume it looks like different things to different people.

**Comment by jacobdadams** [comment]
Agreed, a larger discussion is definitely a good thing. I just had this thought today while we were talking with Miriam (the intern we're getting to tackle metadata) and figured github was a better place to put it down so others could see it than some text file that I'll forget about tomorrow.
